src/main/java/io/nats/client/impl/OrderedPullMessageManager.java

```diff
@@ -57,7 +57,12 @@ protected ManageResult manage(Message msg) {
         if (msg.isJetStream()) {
             long receivedConsumerSeq = msg.metaData().consumerSequence();
             if (expectedExternalConsumerSeq != receivedConsumerSeq) {
-                handleErrorCondition();
+                targetSid.set(null);
+                expectedExternalConsumerSeq = 1; // consumer always starts with consumer sequence 1
+                resetTracking();
+                if (pullManagerObserver != null) {
+                    pullManagerObserver.heartbeatError();
+                }
```

Comment:
this is just simplified - the manager doesn't really need to do all this stuff like it did with push ordered. Now we just let the observer know there was a problem and it handles it.